src/Security/RootPermissions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace SilverStripe\Security;
+
+/**
+ * For root objects that have permission type "Inherit" determine the result of certain permission checks
```

Comment:
I had a hard time understanding the purpose of this class. Suggested alternative:

> Allows objects to enforce permissions for the "root" level, where permissions can not be tied to a particular database record. Objects below the "root" level should use their own can*() implementations instead of this interface.